### ISSUE TITLE
Documentation for quiver.core.firstNonNull contains an extra square brakets

### ISSUE THREAD

**Issue by jeffque** [google/quiver-dart#575]
The documentation for the function `quiver.core.firstNonNull` is being displayed like this:

```dart
firstNonNull(dynamic o1, [ dynamic o2, [ dynamic o3 dynamic o4 ])
```

with an extra square brakets between `o1` and `o2`.

> I also noted a lack of comma separating `o3` and `o4` now. Maybe related?

It can be found at least [here](https://pub.dev/documentation/quiver/latest/quiver.core/firstNonNull.html) and [here](https://pub.dev/documentation/quiver/latest/quiver.core/quiver.core-library.html)

But, seeing [the code](https://github.com/google/quiver-dart/blob/a5412e39941343a931e988bee72b8690cc5ed809/lib/core.dart#L27), I would expect something like this

```dart
firstNonNull(dynamic o1, dynamic o2, [ dynamic o3, dynamic o4 ])
```

or this:

```dart
firstNonNull(o1, o2, [ o3, o4 ])
```

**Comment by cbracken** [comment]
Thanks for filing this!

This looks like a bug in dartdoc as opposed to with quiver itself, so I'll close this and open a bug in https://github.com/dart-lang/dartdoc. 

/cc @jcollins-g are you the right person to follow up with Dartdoc issues these days?

**Comment by cbracken** [comment]
Unrelated to the documentation issue, but for what its worth, `firstNonNull` was authored before Dart supported null-aware operators such as `??`. We should probably deprecate it, and users should replace usages with `o1 ?? o2 ?? o3 ?? o4` etc.

**Comment by jeffque** [comment]
Yay, dartdoc issue!

Actually, I suspected that was some tool bug, but as someone new to the Dart/Flutter environment, I had no idea what tool.

For the `firstNonNull`, I was expecting to meet an ellipsis or something as is in Java, but then I've found it was a upto 4 parameter function.

Thanks for the feedback

**Comment by cbracken** [comment]
No problem! Dart doesn't (yet?) have varargs support, so for the time being, it's a matter of tacking on as many extra optional parameters as you think someone might reasonably use.

**Comment by cbracken** [comment]
FYI dart-lang/dartdoc#2059 was fixed just now via dart-lang/dartdoc#2075. This should land in one of the next couple stable Dart SDK releases. Thanks again for reporting this!
